http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -726,16 +722,41 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift: Parallel](
       task = for {
         _ <- parPreAppCreationSetup
         // create and monitor app
-        _ <- gkeInterp.createAndPollApp(CreateAppParams(msg.appId, msg.project, msg.appName)).adaptError {
-          case e =>
-            PubsubKubernetesError(
-              AppError(e.getMessage, ctx.now, ErrorAction.CreateGalaxyApp, ErrorSource.App, None),
-              Some(msg.appId),
-              false,
-              None,
-              None
-            )
-        }
+        _ <- gkeInterp
+          .createAndPollApp(CreateAppParams(msg.appId, msg.project, msg.appName))
+          .onError {
+            case _ =>
+              // clean-up resources in the event of an app creation error
+              for {
+                _ <- logger.info(
+                  s"Attempting to clean up resources due to app creation error for app ${msg.appName} in project ${msg.project}. | trace id: ${ctx.traceId}"
+                )
+                // we need to look up the app because we always want to clean up the nodepool associated with an errored app, even if it was pre-created
+                appOpt <- KubernetesServiceDbQueries.getFullAppByName(msg.project, msg.appId).transaction
+                // note that this will only clean up the disk if it was created as part of this app creation.
+                // it should not clean up the disk if it already existed
+                _ <- appOpt.traverse { app =>
+                  val deleteMsg =
```

Comment:
One comment: I think `GKEInterpreter. deleteAndPollApp` will set the app status to `Deleted`. We probably want to leave it in `Error` status in this case, so the user can view the error and choose to delete the Error'd app themselves. Maybe GKEInterpreter.deleteAndPollApp` should take a parameter or something to tell it whether to move to `Deleted` status.

Reply:
Good point. Justin and I had briefly touched on this when the issue of [recovering from cluster creation errors](https://broadworkbench.atlassian.net/browse/IA-2197) came up. As discussed there, I wonder if we should use a different status than `ERROR`/`DELETED` so offline clean-up jobs would know how to handle it? Or is this not necessary because resources in `ERROR` status (as far as Leo is concerned) but `DELETED` in Google are already properly handled.

Reply:
is there a case where Leonardo care when a resource is in Error and it's `Deleted` in google, vs Error + still exists in google? To be clear, there should be only `Error` +`Deleted`, and `Error` + `Error` (`Error` + `Running` would be illegal)

Reply:
Well our current pattern when an error occurs is to delete all underlying resources in Google but keep the record in Leo in Error status. That way the user can see that an error occurred, but are not spending money. Agree that the resource-validator will need to handle this situation but I feel like we can achieve this without the need for an additional status code (unless I'm missing something).

Reply:
I think currently this is the only pathway to `Error` -- so we only have `Error` + `Deleted`, there is no such thing (atm) as `Error` + `Error` for k8s apps.

So I feel like we don't need another status code at this time, maybe we can add one when there are more pathways to get an app into `Error` status.

Reply:
yea, to clarify, I don't think there's a need to introduce an additional status either...Just want to ask the question in case I'm missing something

Reply:
Sounds like there is nothing new here that we aren't already handling as part of `Error` + `Deleted` scenarios 👍

Reply:
So, deleteAndPollApp will set the app status to deleted. However, since an error occurred, this error will bubble up and the app will be marked as Error in the top-level handler.

There is a test to confirm this https://github.com/DataBiosphere/leonardo/pull/1601/files#diff-8e28b041fa5855e1d9f7498b905ed3f4R1469

Reply:
I thought of that but would that still result in `destroyedDate` being set? I think the end result should be that `destroyedDate` is _unset_ so the app is still queryable via GET endpoints

Reply:
Yes, the destroyed date will be set. The alternative strikes me as a bit odd though, because it is indeed deleted in google and otherwise the user has to explicitly clean up the app in Leo. Its kind of like we are lying about it still existing for the purpose of showing them an error happened, while the list endpoint can still show it via the `includeDeleted` flag if it is desired. 

I'll change it just to avoid confusion from the UI team since this is how we currently handle errors.

Reply:
Yeah I see how it seems odd but that's how our queries work today -- e.g. `getActiveApp` queries `WHERE destroyedDate == dummyDate`. So any apps with a set `destroyedDate` are effectively invisible.